Engine/src/Input/GLFWKeyMapper.bf

```diff
@@ -163,8 +179,31 @@ namespace SteelEngine.Input
 				case .Button5: return .Mouse4;
 				case .Button6: return .Mouse5;
 				case .Button7: return .Mouse6;
-	
-				case .Button8: return .None; 	// @TODO
+				case .Button8: return .Mouse7;
+				default: return .None;
+			}
+		}
+
+		public static KeyCode MapGamepadButton(GlfwInput.GamepadButton button)
+		{
+			switch (button)
+			{
+				case .A: return .GamepadA;
+				case .B: return .GamepadB;
+				case .X: return .GamepadX;
+				case .Y: return .GamepadY;
+				case .DPadLeft: return .GamepadLeft;
+				case .DPadRight: return .GamepadRight;
+				case .DPadUp: return .GamepadUp;
+				case .DPadDown: return .GamepadDown;
+				case .LeftThumb: return .GamepadLeftStick;
+				case .RightThumb: return .GamepadRightStick;
+				case .Start: return .GamepadStart;
+				case .Guide: return .GamepadHome;
+				case .Back: return .GamepadSelect;
+				case .LeftBumper: return .GamepadL1;
+				case .RightBumper: return .GamepadR1;
```

Comment:
If we are going to go with `A, B, X, Y`, we should stay consistent with XBOX controller naming standards and keep these as `Back, LeftBumper, RightBumper`.

Reply:
I'm still considering renaming the A, B, X, Y buttons to something more generic like South, West, East, North. But yes it should be more consistent.

Reply:
Many people don't actually know their compass directions. I would advocate for the XBOX controller reference, or `Up, Down, Left, Right`, and change DPad to `DPadUp, DPadDown, DPadLeft, DPadRight`